### PR TITLE
Replace deprecated sycl::ONEAPI namespace w/ sycl::ext::oneapi

### PR DESCRIPTION
**Pull request by joeatodd**
`sycl::ONEAPI` doesn't exist on the tip of the `sycl` branch, it's been replaced with `sycl::ext::oneapi`.

### CODE REVIEW

File: core/src/desul/atomics/Macros.hpp [hunk 1]
```diff
@@ -35,8 +35,8 @@ SPDX-License-Identifier: (BSD-3-Clause)
 #ifdef SYCL_LANGUAGE_VERSION
 #define DESUL_HAVE_SYCL_ATOMICS
 #ifdef __clang__
-#define DESUL_SYCL_NAMESPACE sycl::ONEAPI
-#else
+#define DESUL_SYCL_NAMESPACE sycl::ext::oneapi
+#elseg
```
Comment (masterleinad) [inline core/src/desul/atomics/Macros.hpp:39]:
```suggestion
#else
```

**Review by masterleinad — CHANGES_REQUESTED** [review]
Most of the changes here are already contained in https://github.com/kokkos/kokkos/pull/4244/commits/3c526432246a4480e85bf23eb0934297800e5b4b. I thought about splitting that pull request anyway so I'm fine with updating it here.

Apart from the typo, this looks good to me.

**Comment by masterleinad** [comment]
On second thought, this will probably not pass CI since the compiler version is too old. Maybe we just rebase after #4244.

**Comment by dalg24** [comment]
> On second thought, this will probably not pass CI since the compiler version is too old. Maybe we just rebase after #4244.

I don't understand.  I thought all changes proposed here are included in #4244

**Comment by masterleinad** [comment]
> I don't understand. I thought all changes proposed here are included in #4244

Yes, I think so. There are slight differences (`sycl::ext::oneapi::atomic_fence` instead of `sycl::atomic_fence`) but this pull request should be superseded by #4244.

**Comment by dalg24** [comment]
So close it?

**Comment by masterleinad** [comment]
Let's ask first. @joeatodd does #4244 work for you as a replacement for this one?

**Comment by joeatodd** [comment]
yes, superseded by #4244

**Pull request closed without merging**